keps/sig-auth/2862-fine-grained-kubelet-authz/README.md

```diff
@@ -0,0 +1,1013 @@
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We propose a change to the Kubelet API authorization to provide more
+fine-grained control. Currently, the API uses a coarse authorization scheme,
+where actions like reading health status and the ability to exec into a pod
+require the same permission.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The Kubelet API on the authenticated(10250) port currently uses a very coarse
+scheme for authorizing requests. For example, reading `/healthz` and calling
+`/exec/…` (i.e. execute arbitrary code) require the same `proxy` subresource.
+So, if an application needs to say `list` `pods` on the node, as many monitoring
+and logging agents do, they must be granted the `proxy` subresource in RBAC.
+Doing so grants many other powerful permissions to these agents which could be
+exploited by an attacker to escalate privilege.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Introduce new subresources for `/healthz` and `/pods/` kubelet endpoints,
+allowing for more  granular authorization.
+- These changes should be backwards compatible.
+- These changes should not break users on upgrade.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Create a new Kubelet API.
+- Remove the existing subresources.
+- Make the kubelet API node-restricted.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Add new subresources for `healthz` and `pods` endpoint while supporting the old coarse grained
+proxy subresources as shown in the table below.
```

Comment:
"subresource" is a little overloaded here. These are represented as subresources in the `SubjectAccessReview`, but aren't really subresources in the sense of the Kubernetes REST API. To disambiguate, consider referring to these as "authorization subresource".

Reply:
updated subresource with "authorization subresource". Good call!